pkg/safeguards/kustomize-helm-helpers.go

```diff
@@ -0,0 +1,110 @@
+package safeguards
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/chartutil"
+)
+
+// DirectoryType represents the type of directory.
+type DirectoryType int
+
+const (
+	Unknown   DirectoryType = iota // Default value if type cannot be determined
+	Helm                           // Represents a Helm chart directory
+	Kustomize                      // Represents a Kustomize configuration directory
+)
+
+func CreateTempDir() (string, error) {
+	log.Infof("Creating temporary directory for rendered kustomize/helm manifests")
+	tempDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		log.Errorf("Error creating temporary directory to store rendered kustomize/helm manifests: %s", err)
+		return "", fmt.Errorf("failed to create temp directory: %w", err)
+	}
+
+	return tempDir, nil
+}
+
+func GetDirectoryType(dirPath string) DirectoryType {
+	if isHelm, _ := isHelmChart(dirPath); isHelm {
+		return Helm
+	}
+	// } else if isKustomizeConfig(dirPath) {
+	// 		return Kustomize
+	// }
+
+	return Unknown
+}
+
+func isHelmChart(chartDir string) (bool, error) {
+	// Check if the directory exists
+	if _, err := os.Stat(chartDir); os.IsNotExist(err) {
+		return false, fmt.Errorf("directory does not exist: %s", chartDir)
+	}
+
+	// // Load the chart
+	// _, err := loader.Load(chartDir)
+	// if err != nil {
+	// 	return fmt.Errorf("failed to load chart: %s", err)
+	// }
+
+	// Validate the chart
+	if isHelm, err := chartutil.IsChartDir(chartDir); err != nil {
+		return isHelm, fmt.Errorf("chart validation failed: %s", err)
+	}
+
+	return true, nil
+}
+
+// isHelmChart checks if the given directory is a valid Helm chart with required components.
+func isHelmChart2(chartDir string) error {
```

Comment:
same as above

Reply:
This was just a tester function, removing it since I found a library that checks if a directory is a valid helm chart instead of me checking for Chart, values, and templates manually